### DISCUSSION TITLE
Number of days

### DISCUSSION THREAD

**Discussion by fpsilva-source** [rod-trent/MustLearnKQL#26] [Q&A]
One question, do you know how I transform a date into a number of days.

Creation date - current date = number of days.

**Comment by qc-gordon** [comment]
Does this help? The variable `t` is a [timespan](https://docs.microsoft.com/en-us/azure/data-explorer/kusto/query/scalar-data-types/timespan) and then to get the number of days you just divide the time period by `time(1d)`

```
| extend t=now()-TimeGenerated
| extend d=t/time(1d)
```

**Comment by fpsilva-source** [comment]
Thanks @qc-gordon ....

**Reply by CliveWatsonQC** [reply]
There is also: https://docs.microsoft.com/en-gb/azure/data-explorer/kusto/query/datetime-difffunction

e.g.

| extend days_ = datetime_diff('day', max(TimeGenerated), min(TimeGenerated))
